### PR TITLE
Introduce crud.min and crud.max

### PR DESCRIPTION
**Pull request by dokshina**
I didn't forget about

- [x] Tests
- [x] Changelog
- [x] Documentation

### CODE REVIEW

File: crud/borders.lua [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        return true
+    end
+
+    local cmp = keydef:compare(tuple, res_tuple)
+
+    return border_name == 'min' and cmp < 0 or border_name == 'max' and cmp > 0
+end
+
+local function get_border(border_name, space_name, index_name, opts)
+    checks('string', 'string', '?string', {
```
Comment (olegrok) [inline crud/borders.lua:60]: `index_name` could be also a number.

Usually I use `space.index[0]:min()`. I don't use index name but I know that 0 is a primary key.

File: crud.lua [hunk 2]
```diff
@@ -69,6 +70,9 @@ crud.unflatten_rows = utils.unflatten_rows
 -- @function truncate
 crud.truncate = truncate.call
 
+crud.min = borders.min
+crud.max = borders.max
```
Comment (olegrok) [inline crud.lua:74]: Where 

```
-- @refer borders.min
-- @function min
```

File: crud/borders.lua [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+local BorderError = errors.new_class('Border',  {capture_stack = false})
+
+local borders = {}
+
+local STAT_FUNC_NAME = '_crud.get_border_on_storage'
+
+
+local function get_border_on_storage(border_name, space_name, index_id, field_names)
+    dev_checks('string', 'string', 'number', '?table')
```
Comment (olegrok) [inline crud/borders.lua:19]: index_id could be string or number

File: crud/borders.lua [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+    local function get_border(index)
+        return index[border_name](index)
+    end
+
+    return schema.wrap_func_result(space, get_border, {index}, {
+        add_space_schema_hash = true,
+        field_names = field_names,
+    })
+end
```
Comment (olegrok) [inline crud/borders.lua:41]: Is it tested anyhow?

File: crud/borders.lua (hunk 3 above)
Comment (dokshina) [reply crud/borders.lua:19]: Here - no. See `get_border` function on router.

File: crud/borders.lua [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+    for _, result in pairs(results) do
+        if result[1].err ~= nil then
+            return nil, BorderError:new("Failed to get %s: %s", border_name, result.err)
+        end
+
+        local tuple = result[1].res
+        if tuple ~= nil then
+            table.insert(tuples, tuple)
+        end
+    end
```
Comment (olegrok) [inline crud/borders.lua:116]: I suppose you can calculate min/max without additional loop and temporary `tuples` table.

File: crud/borders.lua [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+    if space == nil then
+        return nil, BorderError:new("Space %q doesn't exist", space_name), true
+    end
+
+    local index
+    if index_name == nil then
+        index = space.index[0]
+    else
+        index = space.index[index_name]
+    end
```
Comment (olegrok) [inline crud/borders.lua:77]: Why can't you pass index_name as is to storage?

File: crud/borders.lua (hunk 4 above)
Comment (dokshina) [reply crud/borders.lua]: Added tests.

**Review by olegrok — APPROVED** [review]

**Review by olegrok — APPROVED** [review]
Five stars

**Pull request merged**